CppSource/Source/Services/Stats/Manager/stats_manager.cpp

```diff
@@ -146,24 +176,25 @@ StatsManagerGetStatNames(
 try
 {
     verify_global_init();
+    auto singleton = get_singleton_for_stats();
 
-    statsVars.cppStatNameList.clear();
-    statsVars.cppVoidResult = stats_manager::get_singleton_instance()->get_stat_names(user->pImpl->cppUser(), statsVars.cppStatNameList);
+    singleton->m_statsVars->cppStatNameList.clear();
```

Comment:
might make sense to save multiple of these as well, but this is probably fine for now

Reply:
For C# it shouldn't matter since this data isn't referenced after marshalling. Would have to store each iteration for a cpp layer though.

---
In reply to: [157636754](https://github.com/Microsoft/xbox-live-unity-plugin/pull/181#discussion_r157636754) [](ancestors = 157636754)

Reply:
This should return the same values every time, unless a new stat is published while the title is running. For this type of behavior I don't see why we would need to store multiple of these.

---
In reply to: [157638274](https://github.com/Microsoft/xbox-live-unity-plugin/pull/181#discussion_r157638274) [](ancestors = 157638274,157636754)